third_party/move/move-vm/runtime/src/native_functions.rs

```diff
@@ -195,26 +200,40 @@ impl<'a, 'b, 'c> NativeContext<'a, 'b, 'c> {
 
     pub fn load_function(
         &mut self,
-        module: &ModuleId,
+        module_id: &ModuleId,
         function_name: &Identifier,
     ) -> PartialVMResult<Arc<Function>> {
-        // Load the module that contains this function regardless of the traversal context.
-        //
-        // This is just a precautionary step to make sure that caching status of the VM will not alter execution
-        // result in case framework code forgot to use LoadFunction result to load the modules into cache
-        // and charge properly.
-        self.resolver
-            .loader()
-            .load_module(module, self.data_store, self.resolver.module_store())
-            .map_err(|_| {
-                PartialVMError::new(StatusCode::FUNCTION_RESOLUTION_FAILURE)
-                    .with_message(format!("Module {} doesn't exist", module))
-            })?;
-
-        let (_, function) = self
-            .resolver
-            .module_store()
-            .resolve_module_and_function_by_name(module, function_name)?;
+        let (_, function) = match self.resolver.loader() {
+            Loader::V1(loader) => {
+                // Load the module that contains this function regardless of the traversal context.
+                //
+                // This is just a precautionary step to make sure that caching status of the VM will not alter execution
+                // result in case framework code forgot to use LoadFunction result to load the modules into cache
+                // and charge properly.
+                loader
+                    .load_module(module_id, self.data_store, self.resolver.module_store())
+                    .map_err(|_| {
+                        PartialVMError::new(StatusCode::FUNCTION_RESOLUTION_FAILURE)
+                            .with_message(format!("Module {} doesn't exist", module_id))
+                    })?;
+
+                self.resolver
+                    .module_store()
+                    .resolve_module_and_function_by_name(module_id, function_name)?
+            },
+            Loader::V2(loader) => loader
+                .load_function_without_ty_args(
+                    self.resolver.module_storage(),
+                    module_id.address(),
+                    module_id.name(),
+                    function_name,
+                )
+                // Note: Keeping this consistent with Loader V1 implementation.
```

Comment:
I like the note, but would add more context, as the V1 code above will be removed.

Reply:
Added, also changed to TODO, because not remapping the error breaks some tests, but replay might be ok